{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "homeDir = os.getcwd()+\"/\"\n",
    "fileDir = homeDir+\"Analysis/\"\n",
    "workDir = \"/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: \n",
      "   *****       ***   vcfR   ***       *****\n",
      "   This is vcfR 1.12.0 \n",
      "     browseVignettes('vcfR') # Documentation\n",
      "     citation('vcfR') # Citation\n",
      "   *****       *****      *****       *****\n",
      "\n",
      "\n",
      "R[write to console]: Loading required package: foreach\n",
      "\n",
      "R[write to console]: Loading required package: iterators\n",
      "\n",
      "R[write to console]: Loading required package: parallel\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 306237\n"
     ]
    }
   ],
   "source": [
    "%%R -i fileDir -i workDir \n",
    "#devtools::install_github(repo=\"knausb/vcfR\")\n",
    "#install.packages('doParallel')\n",
    "library(vcfR)\n",
    "library(doParallel)\n",
    "#setwd(\"/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/\")\n",
    "dir.create(fileDir)\n",
    "setwd(workDir)\n",
    "\n",
    "# VCF-------------------\n",
    "vcfDir <- \"ensemble/vcf/\"\n",
    "filenames <- dir(vcfDir,\"ensemble.snpEff.vcf.gz\")\n",
    "#filenames <- filenames[!grepl(\".tbi\",filenames)]\n",
    "\n",
    "vcfRead <- function(filename,vcfDir){\n",
    "  tumorID <- sub(\"-ensemble.snpEff.vcf.gz\",\"\",filename)\n",
    "  tmp <- vcfR::read.vcfR(paste0(vcfDir,filename), verbose=FALSE)\n",
    "  tmpVcf <- tmp@fix\n",
    "  tmpGt <- tmp@gt\n",
    "  if(length(colnames(tmpGt))==3){\n",
    "    if(!grepl(\"_N\\\\b\",colnames(tmpGt)[3])){ #CHECK THE NAME OF TUMOR AND NORMAL SAMPLES IN VCF!\n",
    "      tmpGt <- tmpGt[,c(1,3,2)]\n",
    "    }\n",
    "  }\n",
    "  info <- tmpVcf[,\"INFO\"]\n",
    "  func <- sapply(info,function(x){\n",
    "    temp <- unlist(strsplit(x,\",\"))\n",
    "    paste0(unique(na.omit(unlist(sapply(temp,function(x) unlist(strsplit(x,\"|\",fixed=TRUE))[2])))),collapse=\",\")\n",
    "  })\n",
    "  func <- unname(func)\n",
    "  callers <- sapply(info,function(x){\n",
    "    temp <- unlist(strsplit(x,\";\"))\n",
    "    sub(\"CALLERS=\",\"\",temp[grepl(\"CALLERS=\",temp)])\n",
    "  })\n",
    "  callers <- unname(callers)\n",
    "  id <- paste0(tumorID,\"-\",tmpVcf[,1],\":\",tmpVcf[,2])\n",
    "  if(length(colnames(tmpGt))==3){\n",
    "    tmp <- data.frame(tmpVcf[,c(1,2,4,5,7)],tmpGt,tumorID,info,func,callers,id,stringsAsFactors=FALSE)\n",
    "  } else {\n",
    "    tmp <- data.frame(tmpVcf[,c(1,2,4,5,7)],tmpGt,normal=NA,tumorID,info,func,callers,id,stringsAsFactors=FALSE)\n",
    "  }\n",
    "  colnames(tmp) <- c(\"CHROM\",\"POS\",\"REF\",\"ALT\",\"FILTER\",\"FORMAT\",\"Tumor\",\"Normal\",\"Tumor_ID\",\"snpEff\",\"snpEff_func\",\"caller\",\"id\")\n",
    "  rownames(tmp) <- NULL\n",
    "  return(tmp)\n",
    "}\n",
    "\n",
    "vcf <- lapply(filenames,function(x) vcfRead(x,vcfDir))\n",
    "vcf <- do.call(rbind,vcf)\n",
    "nrow(vcf) #306237\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CORRECT VCF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R \n",
    "# CORRECT FOR NA ALT\n",
    "table(is.na(vcf$ALT))\n",
    "# vcf <- vcf[!is.na(vcf$ALT),]\n",
    "\n",
    "# CORRECT FOR MULTIPLE ALT\n",
    "table(grepl(\",\",vcf$ALT))\n",
    "tmp <- sapply(vcf$ALT,function(x) unlist(strsplit(x,\",\",fixed=TRUE))[1])\n",
    "vcf$ALT <- tmp\n",
    "\n",
    "# # CORRECT FOR NA LINES\n",
    "#sum(is.na(vcf))\n",
    "#vcf <- vcf[!is.na(vcf)]\n",
    "#sum(is.na(vcf))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## SELECT CODING/SPLICING"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R \n",
    "# SELECT CODING/SPLICING\n",
    "tmp <- unique(vcf$snpEff_func)\n",
    "tmp <- unique(unlist(sapply(tmp,function(x) unlist(strsplit(x,\",\")))))\n",
    "tmp <- unique(unlist(sapply(tmp,function(x) unlist(strsplit(x,\"&\")))))\n",
    "tmp <- tmp[!grepl(\"ENS\",tmp)]\n",
    "tmp <- tmp[!grepl(\"=\",tmp)]\n",
    "sort(tmp)\n",
    "sel <- c(\"5_prime_UTR_premature_start_codon_gain_variant\",\"5_prime_UTR_truncation\", #NEW\n",
    "         \"3_prime_UTR_variant\", \"5_prime_UTR_variant\", # Added\n",
    "         \"bidirectional_gene_fusion\",\"conservative_inframe_deletion\",\"conservative_inframe_insertion\",\n",
    "         \"disruptive_inframe_deletion\",\"disruptive_inframe_insertion\",\n",
    "         \"exon_loss_variant\",\"initiator_codon_variant\",\"gene_fusion\",\n",
    "         \"frameshift_variant\",\"missense_variant\",\"splice_acceptor_variant\",\n",
    "         \"splice_donor_variant\",\"splice_region_variant\",\"start_lost\",\"stop_gained\",\n",
    "         \"stop_lost\",\"stop_retained_variant\")\n",
    "setdiff(tmp,sel)\n",
    "setdiff(sel,tmp)\n",
    "sel <- sapply(sel,function(x) grep(x,vcf$snpEff_func))\n",
    "sel <- unique(unlist(sel))\n",
    "sel <- sort(sel)\n",
    "sum(is.na(sel))\n",
    "nrow(vcf) #227168\n",
    "rownames(vcf) <- NULL\n",
    "length(sel) #36938\n",
    "vcf <- vcf[sel,]\n",
    "sort(table(vcf$Tumor_ID))\n",
    "\n",
    "callerN <- sapply(vcf$caller,function(x) length(unlist(strsplit(x,\",\",fixed=TRUE))))\n",
    "callerN <- unname(callerN)\n",
    "vcf <- data.frame(vcf,callerN,stringsAsFactors = FALSE)\n",
    "table(vcf$callerN)\n",
    "#   1     2     3     4     5     6 \n",
    "# 29690  3753   704   328   438  2025  \n",
    "rownames(vcf) <- NULL\n",
    "#----------------------------------------------\n",
    "\n",
    "save(vcf,file = paste(fileDir,\"vcf.rda\",sep=\"\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## COUNTS and ALLELE FREQUENCY"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R \n",
    "# COUNTS and ALLELE FREQUENCY -----------------\n",
    "\n",
    "countsVcf <- function(vcf,i){\n",
    "  Tmp <- rep(NA,4)\n",
    "  names(Tmp) <- c(\"t_ref_count\",\"t_alt_count\",\"n_ref_count\",\"n_alt_count\")\n",
    "  vcfLine <- as.character(vcf[i,])\n",
    "  names(vcfLine) <- colnames(vcf)\n",
    "  caller <- unlist(strsplit(vcfLine[\"caller\"],\",\",fixed=TRUE))[1]\n",
    "  if(caller==\"varscan\"){\n",
    "    tmp <- unlist(strsplit(vcfLine[\"Tumor\"],\":\"))\n",
    "    names(tmp) <- unlist(strsplit(vcfLine[\"FORMAT\"],\":\"))\n",
    "    Tmp[\"t_alt_count\"] <- as.numeric(tmp[\"AD\"])\n",
    "    Tmp[\"t_ref_count\"] <- as.numeric(tmp[\"RD\"])\n",
    "    tmp <- unlist(strsplit(vcfLine[\"Normal\"],\":\"))\n",
    "    names(tmp) <- unlist(strsplit(vcfLine[\"FORMAT\"],\":\"))\n",
    "    Tmp[\"n_alt_count\"] <- as.numeric(tmp[\"AD\"])\n",
    "    Tmp[\"n_ref_count\"] <- as.numeric(tmp[\"RD\"])\n",
    "  } else {\n",
    "    tmp <- unlist(strsplit(vcfLine[\"Tumor\"],\":\"))\n",
    "    names(tmp) <- unlist(strsplit(vcfLine[\"FORMAT\"],\":\"))\n",
    "    if(tmp[\"AD\"]!=\".\") Tmp[c(\"t_ref_count\",\"t_alt_count\")] <- as.numeric(unlist(strsplit(tmp[\"AD\"],\",\"))[1:2])\n",
    "    if(!is.na(vcfLine[\"Normal\"])) {\n",
    "      tmp <- unlist(strsplit(vcfLine[\"Normal\"],\":\"))\n",
    "      names(tmp) <- unlist(strsplit(vcfLine[\"FORMAT\"],\":\"))[1:length(tmp)]\n",
    "      if(tmp[\"AD\"]!=\".\") Tmp[c(\"n_ref_count\",\"n_alt_count\")] <- as.numeric(unlist(strsplit(tmp[\"AD\"],\",\"))[1:2])\n",
    "    }\n",
    "  }\n",
    "  return(Tmp)\n",
    "}\n",
    "\n",
    "cl <- makePSOCKcluster(60,outfile=\"\")\n",
    "registerDoParallel(cl)\n",
    "tmp <- foreach(i = 1:nrow(vcf)) %dopar% {\n",
    "  tmp <- countsVcf(vcf,i)\n",
    "  return(tmp)\n",
    "}\n",
    "stopCluster(cl)\n",
    "\n",
    "tmp <- do.call(rbind,tmp)\n",
    "dim(tmp) #36938     4\n",
    "\n",
    "counts <- cbind(tmp[,1:2],NA,NA,tmp[,3:4],NA,NA)\n",
    "colnames(counts) <- c(\"t_ref_count\",\"t_alt_count\",\"t_depth\",\"t_vaf\",\"n_ref_count\",\"n_alt_count\",\"n_depth\",\"n_vaf\")\n",
    "counts[,\"t_depth\"] <- counts[,\"t_alt_count\"] + counts[,\"t_ref_count\"]\n",
    "counts[,\"t_vaf\"] <- round(counts[,\"t_alt_count\"]/counts[,\"t_depth\"],2)\n",
    "counts[,\"n_depth\"] <- counts[,\"n_alt_count\"] + counts[,\"n_ref_count\"]\n",
    "counts[,\"n_vaf\"] <- round(counts[,\"n_alt_count\"]/counts[,\"n_depth\"],2)\n",
    "rownames(counts) <- NULL\n",
    "\n",
    "#View(counts[sample(1:nrow(counts),100),])\n",
    "\n",
    "vcf <- data.frame(vcf, counts, stringsAsFactors = FALSE)\n",
    "\n",
    "table(cond <- is.na(vcf$n_depth)&!is.na(vcf$Normal))\n",
    "# FALSE  TRUE \n",
    "# 27894  9044 \n",
    "# table(cond <- is.na(vcf$n_depth))\n",
    "\n",
    "tmp <- vcf[cond,]\n",
    "\n",
    "# table(tmp$caller)\n",
    "# freebayes \n",
    "# 9044\n",
    "\n",
    "vcf <- vcf[!cond,]\n",
    "\n",
    "rownames(vcf) <- NULL\n",
    "nrow(vcf) #27894\n",
    "rm(counts)\n",
    "\n",
    "#----------------------------------------------\n",
    "# save(vcf,file = \"Analysis/vcf.rda\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## FP FILTER"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "fpfilterDir = fileDir + \"fpfilter/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"A1783\"\n",
      "[1] \"A9155\"\n",
      "[1] \"A9878\"\n",
      "[1] \"AT2550\"\n",
      "[1] \"AT2822\"\n",
      "[1] \"AT4415\"\n",
      "[1] \"AT4808\"\n",
      "[1] \"C0288\"\n",
      "[1] \"C0334\"\n",
      "[1] \"C1572\"\n",
      "[1] \"C8448\"\n",
      "[1] \"s1_3076_D\"\n",
      "[1] \"s1_3295_D\"\n",
      "[1] \"s1_8588_D\"\n",
      "[1] \"s2_6088_D\"\n",
      "[1] \"s2_6339_D\"\n",
      "[1] \"s2_8811_D\"\n",
      "[1] \"s3_5832_D\"\n",
      "[1] \"s3_7299_D\"\n",
      "[1] \"s3_7979_D\"\n"
     ]
    }
   ],
   "source": [
    "%%R -i fpfilterDir\n",
    "# FP FILTER\n",
    "#system(\"mkdir Analysis/fpfilter\")\n",
    "#setwd(\"Analysis/fpfilter/\")\n",
    "dir.create(fpfilterDir)\n",
    "setwd(fpfilterDir)\n",
    "\n",
    "vcf <- data.frame(vcf,fpFid=\"\",stringsAsFactors=FALSE)\n",
    "patients <- unique(vcf$Tumor_ID)\n",
    "for(i in 1:length(patients)){\n",
    "  print(patients[i])\n",
    "  name <- patients[i]\n",
    "  var <- vcf[vcf$Tumor_ID==name,1:4]\n",
    "  var[nchar(var$ALT)>1,\"ALT\"] <- paste0(\"+\",substring(var[nchar(var$ALT)>1,\"ALT\"],2))\n",
    "  var[nchar(var$REF)>1,\"POS\"] <- as.numeric(var[nchar(var$REF)>1,\"POS\"])+1\n",
    "  var[nchar(var$REF)>1,\"ALT\"] <- paste0(\"-\",substring(var[nchar(var$REF)>1,\"REF\"],2))\n",
    "  var[nchar(var$REF)>1,\"REF\"] <- substring(var[nchar(var$REF)>1,\"REF\"],2,2)\n",
    "  vcf[vcf$Tumor_ID==name,\"fpFid\"] <- paste0(name,\"-\",var[,1],\":\",var[,2],\"-\",var[,4])\n",
    "  write.table(var,file=paste0(name,\".var\"),sep=\"\\t\",quote=FALSE,row.names=FALSE,col.names=FALSE)\n",
    "  # n <- ceiling(nrow(var)/60)\n",
    "  # system(paste0(\"split -l \",n,\" --additional-suffix .var \",paste0(name,\".var\")))\n",
    "  loc <- paste(var$CHROM,var$POS,var$POS)\n",
    "  write.table(loc,file=paste0(name,\".loc\"),quote=FALSE,row.names=FALSE,col.names=FALSE)\n",
    "  # system(paste0(\"split -l \",n,\" --additional-suffix .loc \",paste0(name,\".loc\")))\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir(fpfilterDir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A1783.loc\n",
      "A1783\n",
      "A1783 bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_Flank/ABT414_Flank/A1783_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/A1783_T_recal.bam\n",
      "\\n\n",
      "A9155.loc\n",
      "A9155\n",
      "A9155 bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_Flank/ABT414_Flank/A9155_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/A9155_T_recal.bam\n",
      "\\n\n",
      "A9878.loc\n",
      "A9878\n",
      "A9878 bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_Flank/ABT414_Flank/A9878_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/A9878_T_recal.bam\n",
      "\\n\n",
      "AT2550.loc\n",
      "AT2550\n",
      "AT2550 bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_Flank/ABT414_Flank/AT2550_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/AT2550_T_recal.bam\n",
      "\\n\n",
      "AT2822.loc\n",
      "AT2822\n",
      "AT2822 bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_Flank/ABT414_Flank/AT2822_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/AT2822_T_recal.bam\n",
      "\\n\n",
      "AT4415.loc\n",
      "AT4415\n",
      "AT4415 bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_Flank/ABT414_Flank/AT4415_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/AT4415_T_recal.bam\n",
      "\\n\n",
      "AT4808.loc\n",
      "AT4808\n",
      "AT4808 bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_Flank/ABT414_Flank/AT4808_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/AT4808_T_recal.bam\n",
      "\\n\n",
      "C0288.loc\n",
      "C0288\n",
      "C0288 bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_Flank/ABT414_Flank/C0288_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/C0288_T_recal.bam\n",
      "\\n\n",
      "C0334.loc\n",
      "C0334\n",
      "C0334 bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_Flank/ABT414_Flank/C0334_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/C0334_T_recal.bam\n",
      "\\n\n",
      "C1572.loc\n",
      "C1572\n",
      "C1572 bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_Flank/ABT414_Flank/C1572_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/C1572_T_recal.bam\n",
      "\\n\n",
      "C8448.loc\n",
      "C8448\n",
      "C8448 bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_Flank/ABT414_Flank/C8448_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/C8448_T_recal.bam\n",
      "\\n\n",
      "s1_3076_D.loc\n",
      "s1_3076_D\n",
      "s1_3076_D bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_IC/ABT414_IC/s1_3076_D_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/s1_3076_D_T_recal.bam\n",
      "\\n\n",
      "s1_3295_D.loc\n",
      "s1_3295_D\n",
      "s1_3295_D bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_IC/ABT414_IC/s1_3295_D_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/s1_3295_D_T_recal.bam\n",
      "\\n\n",
      "s1_8588_D.loc\n",
      "s1_8588_D\n",
      "s1_8588_D bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_IC/ABT414_IC/s1_8588_D_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/s1_8588_D_T_recal.bam\n",
      "\\n\n",
      "s2_6088_D.loc\n",
      "s2_6088_D\n",
      "s2_6088_D bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_IC/ABT414_IC/s2_6088_D_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/s2_6088_D_T_recal.bam\n",
      "\\n\n",
      "s2_6339_D.loc\n",
      "s2_6339_D\n",
      "s2_6339_D bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_IC/ABT414_IC/s2_6339_D_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/s2_6339_D_T_recal.bam\n",
      "\\n\n",
      "s2_8811_D.loc\n",
      "s2_8811_D\n",
      "s2_8811_D bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_IC/ABT414_IC/s2_8811_D_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/s2_8811_D_T_recal.bam\n",
      "\\n\n",
      "s3_5832_D.loc\n",
      "s3_5832_D\n",
      "s3_5832_D bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_IC/ABT414_IC/s3_5832_D_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/s3_5832_D_T_recal.bam\n",
      "\\n\n",
      "s3_7299_D.loc\n",
      "s3_7299_D\n",
      "s3_7299_D bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_IC/ABT414_IC/s3_7299_D_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/s3_7299_D_T_recal.bam\n",
      "\\n\n",
      "s3_7979_D.loc\n",
      "s3_7979_D\n",
      "s3_7979_D bam-readcount\n",
      "/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/ABT414_IC/ABT414_IC/s3_7979_D_T_recal.bam /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/CNV/tumorbam/s3_7979_D_T_recal.bam\n",
      "\\n\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "Minimum mapping quality is set to 0\n",
      "WARNING: In read A00261:111:HFJ5KDSXX:2:1119:8820:27696: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00261:111:HFJ5KDSXX:2:1110:25807:9643: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00261:111:HFJ5KDSXX:2:1453:20573:5149: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00261:111:HFJ5KDSXX:2:1254:23402:13244: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00261:111:HFJ5KDSXX:2:1412:30870:29371: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00261:111:HFJ5KDSXX:2:1342:11595:20541: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00261:111:HFJ5KDSXX:2:2169:31204:21809: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00261:111:HFJ5KDSXX:2:1217:1841:13698: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00261:111:HFJ5KDSXX:2:1465:12662:35008: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00261:111:HFJ5KDSXX:2:2253:4698:31610: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00197:72:HCHTWDSXX:2:1274:26096:16063: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00261:111:HFJ5KDSXX:2:1375:8585:36025: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00197:72:HCHTWDSXX:2:1241:27606:5274: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00197:72:HCHTWDSXX:2:1637:19813:24064: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00197:72:HCHTWDSXX:2:1134:9878:32471: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00197:72:HCHTWDSXX:2:2337:14886:17879: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00197:72:HCHTWDSXX:2:2508:7545:22576: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00197:72:HCHTWDSXX:2:1446:15854:20181: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00197:72:HCHTWDSXX:2:1157:30933:31673: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n",
      "WARNING: In read A00197:72:HCHTWDSXX:2:2660:19226:11804: Couldn't find single-end mapping quality. Check to see if the SM tag is in BAM.\n",
      "The previous warning has been emitted 1 times and will be disabled.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "fasta=/storage/gluster/vol1/bcbio/genomes/Hsapiens/hg19/bwa/hg19.fa\n",
    "bamreadcount=/storage/gluster/vol1/SHARED/NGSTOOLS/bam-readcounts/bam-readcount/build/bin/bam-readcount\n",
    "\n",
    "#cd {fpfilterDir}\n",
    "\n",
    "for file in *.loc\n",
    "do\n",
    "\techo $file\n",
    "\tname=${file%.loc}\n",
    "\techo $name\n",
    "\techo \"$name bam-readcount\"\n",
    "\tbam=`ls /storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/*/*/${name}_T_recal.bam`\n",
    "\techo $bam\n",
    "    echo '\\n'\n",
    "\t$bamreadcount -f $fasta -w 1 -l $file $bam  > ${name}.readcount &\n",
    "done  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A1783.var\n",
      "A9155.var\n",
      "A9878.var\n",
      "AT2550.var\n",
      "AT2822.var\n",
      "AT4415.var\n",
      "AT4808.var\n",
      "C0288.var\n",
      "C0334.var\n",
      "C1572.var\n",
      "C8448.var\n",
      "s1_3076_D.var\n",
      "s1_3295_D.var\n",
      "s1_8588_D.var\n",
      "s2_6088_D.var\n",
      "s2_6339_D.var\n",
      "s2_8811_D.var\n",
      "s3_5832_D.var\n",
      "s3_7299_D.var\n",
      "s3_7979_D.var\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "fpfilter=/storage/gluster/vol1/data/PUBLIC/Tools/VARSCAN/fpfilter.pl\n",
    "\n",
    "for file in *.var\n",
    "do\n",
    "\techo $file\n",
    "\tperl $fpfilter ${file} ${file%.var}.readcount --output-basename ${file%.var}.fpfilter > ${file%.var}.log &\n",
    "done  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 1\n",
      "[1] 2\n",
      "[1] 3\n",
      "[1] 4\n",
      "[1] 5\n",
      "[1] 6\n",
      "[1] 7\n",
      "[1] 8\n",
      "[1] 9\n",
      "[1] 10\n",
      "[1] 11\n",
      "[1] 12\n",
      "[1] 13\n",
      "[1] 14\n",
      "[1] 15\n",
      "[1] 16\n",
      "[1] 17\n",
      "[1] 18\n",
      "[1] 19\n",
      "[1] 20\n",
      "[1] 1\n",
      "[1] 2\n",
      "[1] 3\n",
      "[1] 4\n",
      "[1] 5\n",
      "[1] 6\n",
      "[1] 7\n",
      "[1] 8\n",
      "[1] 9\n",
      "[1] 10\n",
      "[1] 11\n",
      "[1] 12\n",
      "[1] 13\n",
      "[1] 14\n",
      "[1] 15\n",
      "[1] 16\n",
      "[1] 17\n",
      "[1] 18\n",
      "[1] 19\n",
      "[1] 20\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "setwd(fpfilterDir)\n",
    "failfiles <- dir(path=\".\",pattern = \"fpfilter.fail\")\n",
    "FPfail <- NULL\n",
    "for(i in 1:length(failfiles)){\n",
    "  print(i)\n",
    "  patient <- unlist(strsplit(failfiles[i],\".\",fixed=TRUE))[1]\n",
    "  tmp <- readLines(failfiles[i])\n",
    "  tmp <- sapply(tmp,function(x){\n",
    "    Tmp <- unlist(strsplit(x,\"\\t\"))\n",
    "    length(Tmp) <- 9\n",
    "    Tmp\n",
    "  })\n",
    "  tmp <- t(tmp)\n",
    "  failID <- paste0(patient,\"-\",tmp[,1],\":\",tmp[,2],\"-\",tmp[,4])\n",
    "  tmp <- cbind(failID,tmp[,9])\n",
    "  rownames(tmp) <- failID\n",
    "  FPfail <- rbind(FPfail,tmp)\n",
    "}\n",
    "\n",
    "passfiles <- dir(path=\".\",pattern = \"fpfilter.pass\")\n",
    "FPpass <- NULL\n",
    "for(i in 1:length(passfiles)){\n",
    "  print(i)\n",
    "  patient <- unlist(strsplit(passfiles[i],\".\",fixed=TRUE))[1]\n",
    "  tmp <- read.delim(passfiles[i],header=FALSE,as.is=TRUE)\n",
    "  tmp <- paste0(patient,\"-\",tmp[,1],\":\",tmp[,2],\"-\",tmp[,4])\n",
    "  FPpass <- c(FPpass,tmp)\n",
    "}\n",
    "\n",
    "length(FPpass) #8967\n",
    "nrow(FPfail) #18886\n",
    "\n",
    "length(FPpass)+nrow(FPfail) #27853\n",
    "\n",
    "all <- c(FPpass,FPfail[,1])\n",
    "table(vcf$fpFid%in%all)\n",
    "#View(vcf[!vcf$fpFid%in%all,])\n",
    "\n",
    "vcf <- data.frame(vcf,FPfilter=\".\",FPfail=\".\",stringsAsFactors=FALSE)\n",
    "vcf[vcf$fpFid%in%FPpass,\"FPfilter\"] <- \"PASS\"\n",
    "vcf[vcf$fpFid%in%rownames(FPfail),\"FPfilter\"] <- \"FAIL\"\n",
    "vcf[vcf$FPfilter==\"FAIL\",\"FPfail\"] <- FPfail[vcf[vcf$FPfilter==\"FAIL\",\"fpFid\"],2]\n",
    "\n",
    "vcf[is.na(vcf$FPfail),\"FPfail\"] <- \"\"\n",
    "vcf[vcf$FPfail==\"\",\"FPfilter\"] <- \"no_readcounts\" # alt absent in the bam readcount file\n",
    "vcf[vcf$FPfail==\"\",\"FPfail\"] <- \".\"\n",
    "vcf[vcf$FPfilter==\".\",\"FPfilter\"] <- \"no_readcounts\"# alt = 0 in the bam readcount file\n",
    "table(vcf$FPfilter)\n",
    "# FAIL no_readcounts          PASS \n",
    "# 15577          3350          8967  \n",
    "rm(FPpass,FPfail,all)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "# # VirtualNormalCorrection----------\n",
    "# system(\"mkdir VN\")\n",
    "# setwd(\"VN/\")\n",
    "# ID <- 1:nrow(vcf)\n",
    "# VCF <- data.frame(vcf[,1:2],ID,vcf[,3:4],stringsAsFactors = FALSE)\n",
    "# write.table(VCF,sep=\"\\t\",row.names = FALSE,quote=FALSE,file = \"VNinput.vcf\")\n",
    "# n <- ceiling(nrow(VCF)/60)\n",
    "# system(paste0(\"split -l \",n,\" --additional-suffix .vcf VNinput.vcf\"))\n",
    "# \n",
    "# # UNIX\n",
    "# # parallelVN.sh\n",
    "# # parallelVN_combineRes.sh\n",
    "# \n",
    "# VNoutput <- read.delim(\"VNoutput_main.tsv\",as.is=TRUE)\n",
    "# rownames(vcf) <- VCF$ID\n",
    "# vcf <- data.frame(vcf,idNum=VCF$ID,stringsAsFactors = FALSE)\n",
    "# vcf <- data.frame(vcf[VNoutput$xRef,],VNoutput[,9:12],stringsAsFactors = FALSE)\n",
    "# # threshold 5% (~ 22/433) - threshold_highconf 10% (~ 44/433)\n",
    "# table(vcf$VN_occurrences<22&vcf$VN_fullycalled_count>=44)\n",
    "# # FALSE   TRUE \n",
    "# # 220620  84644 \n",
    "# table(vcf$VN_occurrences<3&vcf$VN_fullycalled_count>=110)\n",
    "# # FALSE   TRUE \n",
    "# # 245383  59881 \n",
    "# table(vcf$VN_occurrences<3) # 0.7% (3/433)\n",
    "# # FALSE   TRUE \n",
    "# # 233414  71850 \n",
    "# # table(is.na(vcf$VN_occurrences))\n",
    "# \n",
    "# vcf$vn <- vcf$VN_occurrences<3\n",
    "# table(vcf$vn)\n",
    "# # FALSE   TRUE \n",
    "# # 233414  71850 \n",
    "# \n",
    "# rm(VCF,VNoutput,tmp)\n",
    "\n",
    "setwd(\"/storage/gluster/vol1/data/PUBLIC/SCAMBIO/ABT414_WES_Analysis/Analysis/\")\n",
    "save(vcf,file = paste(fileDir,\"vcf.rda\",sep=\"\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# STEP 2 - ANNOTATION AND DOWNSTREAM ANALYSIS #"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ANNOVAR Annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "\n",
      "R[write to console]: In addition: \n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "R[write to console]: \n",
      " \n",
      "R[write to console]:  cannot open file 'ann.in': Permission denied\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "ann.in <- vcf[,c(1,2,2,3,4)]\n",
    "colnames(ann.in) <- c(\"Chr\",\"Start\",\"End\",\"Ref\",\"Alt\")\n",
    "\n",
    "cond <- nchar(ann.in[,\"Ref\"])>1&nchar(ann.in[,\"Alt\"])==1\n",
    "ann.in[cond,\"Start\"] <- as.numeric(ann.in[cond,\"Start\"])+1\n",
    "ann.in[cond,\"Ref\"] <- substring(ann.in[cond,\"Ref\"],2,nchar(ann.in[cond,\"Ref\"]))\n",
    "ann.in[cond,\"Alt\"] <- \"-\"\n",
    "ann.in[cond,\"End\"] <- as.numeric(ann.in[cond,\"Start\"])+nchar(ann.in[cond,\"Ref\"])-1\n",
    "\n",
    "cond <- nchar(ann.in[,\"Alt\"])>1&nchar(ann.in[,\"Ref\"])==1\n",
    "ann.in[cond,\"Ref\"] <- \"-\"\n",
    "ann.in[cond,\"Alt\"] <- substring(ann.in[cond,\"Alt\"],2,nchar(ann.in[cond,\"Alt\"]))\n",
    "\n",
    "cond <- nchar(ann.in[,\"Ref\"])>1&nchar(ann.in[,\"Alt\"])>1\n",
    "ann.in[cond,\"End\"] <- as.numeric(ann.in[cond,\"Start\"])+nchar(ann.in[cond,\"Ref\"])-1\n",
    "\n",
    "write.table(ann.in,file=\"ann.in\",col.names=FALSE,row.names=FALSE,quote=FALSE,sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "/storage/gluster/vol1/data/PUBLIC/Tools/annovar2017Jul16/table_annovar.pl ann.in \\\n",
    "/storage/gluster/vol1/data/PUBLIC/Tools/annovar2017Jul16/humandb/ -buildver hg19 -out ann.out \\\n",
    "-remove -protocol refGene,avsnp150,snp138NonFlagged,\\\n",
    "exac03,1000g2015aug_all,esp6500siv2_all,kaviar_20150923,hrcr1,\\\n",
    "cosmic80,clinvar_20170905,\\\n",
    "dbnsfp33a,dbscsnv11,dann,eigen,gerp++gt2,cadd  \\\n",
    "-operation g,f,f,f,f,f,f,f,f,f,f,f,f,f,f,f -nastring . --thread 80"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ann.out <- read.delim(\"ann.out.hg19_multianno.txt\",as.is=TRUE)\n",
    "paste0(colnames(ann.out),collapse=\"','\")\n",
    "tmp <- c('Chr','Start','End','Ref','Alt','Func.refGene','Gene.refGene','GeneDetail.refGene','ExonicFunc.refGene','AAChange.refGene', #refGene\n",
    "         'avsnp150', #avsnp150\n",
    "         'snp138NonFlagged', #snp138NonFlagged\n",
    "         'ExAC_ALL','ExAC_AFR','ExAC_AMR','ExAC_EAS','ExAC_FIN','ExAC_NFE','ExAC_OTH','ExAC_SAS', #exac03\n",
    "         'X1000g2015aug_all', #1000g2015aug_all\n",
    "         'esp6500siv2_all', #esp6500siv2_all\n",
    "         'Kaviar_AF','Kaviar_AC','Kaviar_AN', #kaviar_20150923\n",
    "         'HRC_AF','HRC_AC','HRC_AN','HRC_non1000G_AF','HRC_non1000G_AC','HRC_non1000G_AN', #hrcr1\n",
    "         'cosmic80', #cosmic80\n",
    "         'CLINSIG','CLNDBN','CLNACC','CLNDSDB','CLNDSDBID', #clinvar_20170905\n",
    "         'SIFT_score','SIFT_converted_rankscore','SIFT_pred', #dbnsfp33a\n",
    "         'Polyphen2_HDIV_score','Polyphen2_HDIV_rankscore','Polyphen2_HDIV_pred', #dbnsfp33a\n",
    "         'Polyphen2_HVAR_score','Polyphen2_HVAR_rankscore','Polyphen2_HVAR_pred', #dbnsfp33a\n",
    "         'LRT_score','LRT_converted_rankscore','LRT_pred', #dbnsfp33a\n",
    "         'MutationTaster_score','MutationTaster_converted_rankscore','MutationTaster_pred', #dbnsfp33a\n",
    "         'MutationAssessor_score','MutationAssessor_score_rankscore','MutationAssessor_pred', #dbnsfp33a\n",
    "         'FATHMM_score','FATHMM_converted_rankscore','FATHMM_pred', #dbnsfp33a\n",
    "         'PROVEAN_score','PROVEAN_converted_rankscore','PROVEAN_pred', #dbnsfp33a\n",
    "         'VEST3_score','VEST3_rankscore','MetaSVM_score','MetaSVM_rankscore','MetaSVM_pred', #dbnsfp33a\n",
    "         'MetaLR_score','MetaLR_rankscore','MetaLR_pred', #dbnsfp33a\n",
    "         'M.CAP_score','M.CAP_rankscore','M.CAP_pred', #dbnsfp33a\n",
    "         'CADD_raw','CADD_raw_rankscore','CADD_phred', #dbnsfp33a\n",
    "         'DANN_score','DANN_rankscore', #dbnsfp33a\n",
    "         'fathmm.MKL_coding_score','fathmm.MKL_coding_rankscore','fathmm.MKL_coding_pred', #dbnsfp33a\n",
    "         'Eigen_coding_or_noncoding','Eigen.raw','Eigen.PC.raw', #dbnsfp33a\n",
    "         'GenoCanyon_score','GenoCanyon_score_rankscore', #dbnsfp33a\n",
    "         'integrated_fitCons_score','integrated_fitCons_score_rankscore','integrated_confidence_value', #dbnsfp33a\n",
    "         'GERP.._RS','GERP.._RS_rankscore', #dbnsfp33a\n",
    "         'phyloP100way_vertebrate','phyloP100way_vertebrate_rankscore','phyloP20way_mammalian','phyloP20way_mammalian_rankscore', #dbnsfp33a\n",
    "         'phastCons100way_vertebrate','phastCons100way_vertebrate_rankscore','phastCons20way_mammalian','phastCons20way_mammalian_rankscore', #dbnsfp33a\n",
    "         'SiPhy_29way_logOdds','SiPhy_29way_logOdds_rankscore', #dbnsfp33a\n",
    "         'Interpro_domain', #dbnsfp33a\n",
    "         'GTEx_V6_gene','GTEx_V6_tissue', #dbnsfp33a\n",
    "         'dbscSNV_ADA_SCORE','dbscSNV_RF_SCORE', #dbscsnv11 SPLICE SITE\n",
    "         'dann', #dann\n",
    "         'Eigen', #eigen\n",
    "         'gerp..gt2', #gerp++gt2\n",
    "         'CADD','CADD_Phred') #cadd\n",
    "\n",
    "sel <- c('Chr','Start','End','Ref','Alt',\n",
    "         'Func.refGene','Gene.refGene','GeneDetail.refGene','ExonicFunc.refGene','AAChange.refGene', 'Interpro_domain',\n",
    "         'avsnp150','snp138NonFlagged',\n",
    "         'ExAC_ALL','X1000g2015aug_all','esp6500siv2_all','Kaviar_AF','HRC_AF',\n",
    "         'cosmic80','CLINSIG','CLNDBN','CLNACC','CLNDSDB','CLNDSDBID',\n",
    "         'SIFT_pred','Polyphen2_HDIV_pred','MutationTaster_pred','PROVEAN_pred',\n",
    "         'dbscSNV_ADA_SCORE','dbscSNV_RF_SCORE',\n",
    "         'dann','Eigen','gerp..gt2','CADD','CADD_Phred')\n",
    "\n",
    "ann.out <- ann.out[,sel]\n",
    "ann.out[ann.out$ExonicFunc.refGene!=\".\",\"Func.refGene\"] <- ann.out[ann.out$ExonicFunc.refGene!=\".\",\"ExonicFunc.refGene\"]\n",
    "ann.out[ann.out$AAChange.refGene!=\".\",\"GeneDetail.refGene\"] <- ann.out[ann.out$AAChange.refGene!=\".\",\"AAChange.refGene\"]\n",
    "ann.out <- ann.out[,!colnames(ann.out)%in%c(\"ExonicFunc.refGene\",\"AAChange.refGene\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
